src/components/Board.js

```diff
@@ -3,18 +3,29 @@ import './Board.css';
 import Square from './Square';
 import PropTypes from 'prop-types';
 
-
 const generateSquareComponents = (squares, onClickCallback) => {
-  // Complete this for Wave 1
-  // squares is a 2D Array, but 
-  //  you need to return a 1D array
-  //  of square components
+  let onedSquares = []; 
+
+  for (const row of squares) { 
+    for (const square of row) {
+        onedSquares.push(square)
+    }
+  } 
+
+  return onedSquares.map((square) => {
+    return (
+        <Square 
+          value={ square.value }
+          id={ square.id } 
+          onClickCallback={ onClickCallback }
+        />
+    )
+  })
```

Comment:
You can use a few built in methods to clean this up.

You can use `flat()` instead of your first loop:

```suggestion
  let onedSquares = squares.flat();

  return onedSquares.map((square) => {
    return (
        <Square 
          value={ square.value }
          id={ square.id } 
          onClickCallback={ onClickCallback }
        />
    )
  })
```

And then JS actually includes the built in method `flatMap` that does a flat followed by a map all in one step:
```suggestion
  return squares.flatMap((square) => {
    return (
        <Square 
          value={ square.value }
          id={ square.id } 
          onClickCallback={ onClickCallback }
        />
    )
  })
```